tests/components/numato/numato_mock.py

```diff
@@ -0,0 +1,67 @@
+"""Mockup for the numato component interface."""
+from numato_gpio import NumatoGpioError
+
+
+class NumatoModuleMock:
+    """Mockup for the numato_gpio module."""
+
+    NumatoGpioError = NumatoGpioError
+
+    class NumatoDeviceMock:
+        """Mockup for the numato_gpio.NumatoUsbGpio class."""
+
+        def __init__(self, device):
+            """Initialize numato device mockup."""
+            self.device = device
+            self.callbacks = {}
+            self.ports = set()
+            self.values = {}
+
+        def setup(self, port, direction):
+            """Mockup for setup."""
+            self.ports.add(port)
+            self.values[port] = None
+
+        def write(self, port, value):
+            """Mockup for write."""
+            self.values[port] = value
+
+        def read(self, port):
+            """Mockup for read."""
+            return 1
+
+        def adc_read(self, port):
+            """Mockup for adc_read."""
+            return 1023
+
+        def add_event_detect(self, port, callback, direction):
+            """Mockup for add_event_detect."""
+            self.callbacks[port] = callback
+
+        def notify(self, enable):
+            """Mockup for notify."""
+
+        def mockup_inject_notification(self, port, value):
+            """Make the mockup execute a notification callback."""
+            self.callbacks[port](port, value)
+
+    devices = {}
```

Comment:
If we put the `devices` dict as a instance attribute instead and instantiate the mock module class before we use it for patching the gpio module in the fixture we would not need to clean up the fixture in the teardown phase.

It would also be less risky, since mutable objects as class attributes are sneaky.

Reply:
Good idea :-) I felt bad about that mockup but didn't take the time to figure out why and to improve. Thanks!